src/pages/home/report/ReportActionsView.js

```diff
@@ -206,10 +206,16 @@ class ReportActionsView extends React.Component {
             this.openReportIfNecessary();
         }
 
-        // If the report is unread, we want to check if the number of actions has decreased. If so, then it seems that one of them was deleted. In this case, if the deleted action was the
-        // one marking the unread point, we need to recalculate which action should be the unread marker.
-        if (ReportUtils.isUnread(this.props.report) && ReportActionsUtils.filterReportActionsForDisplay(prevProps.reportActions).length > this.sortedAndFilteredReportActions.length) {
-            this.setState({newMarkerReportActionID: ReportUtils.getNewMarkerReportActionID(this.props.report, this.sortedAndFilteredReportActions)});
+        // If the report action marking the unread point is deleted we need to recalculate which action should be the unread marker
+        if (this.state.newMarkerReportActionID) {
+            const reportAction = this.props.reportActions[this.state.newMarkerReportActionID];
+            if (!reportAction || _.isEmpty(lodashGet(reportAction, 'message[0].html')) || reportAction.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
```

Comment:
Report actions are never deleted. We just set the message to an empty string.

Reply:
I think `_.isEmpty(lodashGet(reportAction, 'message[0].html')` may be enough even if `reportAction` is `undefined` (which is a case that I'm not sure if it may even exist). `!reportAction` is only used for an early return. If you want to get rid of that please make a suggestion on that.

Reply:
Updated